holo-layer/tests/src/common.ts

```diff
@@ -4,102 +4,67 @@ import path from "path";
 
 export const conductorConfig = Config.gen({});
 export const Peershare_Zome = "peershare";
-
+export const Default_Developer_Address =
+  "uhCAkbs0mibHm7rB0ZPkZZzhHC55KwmDxt6uR9x2XgFQ82OYvKwR7";
 export const peerShareDNA = path.join(
-    __dirname,
-    "../../workdir/dna/peershare.dna"
+  __dirname,
+  "../../workdir/dna/peershare.dna"
 );
 
 const installationAgents: InstallAgentsHapps = [
-    // agent 0
-    [
-        // happ 0
-        [peerShareDNA],
-    ],
-    [
-        // happ 0
-        [peerShareDNA],
-    ],
+  // agent 0
+  [
+    // happ 0
+    [peerShareDNA],
+  ],
+  [
+    // happ 0
+    [peerShareDNA],
+  ],
 ];
 
-export const InstallAgentApp = async (s, agentName: string) => {
-    const config = Config.gen();
-    const [agent] = await s.players([conductorConfig], false);
-    await agent.startup();
-    const adminWs = agent.adminWs();
-    const agent_key = await adminWs.generateAgentPubKey();
-    const hash_app = await adminWs.registerDna({
-        path: peerShareDNA,
-    });
+import { Base64 } from "js-base64";
+export function serializeHash(hash) {
+  return `u${Base64.fromUint8Array(hash, true)}`;
+}
+export const InstallAgentApp = async (
+  s,
+  agentName: string,
+  am_i_developer: boolean = false
+) => {
+  const config = Config.gen();
+  const [agent] = await s.players([conductorConfig], false);
+  await agent.startup();
+  const adminWs = agent.adminWs();
+  let progenitor: string = Default_Developer_Address;
+  var agent_key = await adminWs.generateAgentPubKey();
+  if (am_i_developer) {
+    progenitor = serializeHash(agent_key);
+  }
+  const hash_app = await adminWs.registerDna({
+    path: peerShareDNA,
+    properties: {
+      developer_address: progenitor,
+    },
+  });
```

Comment:
Ok...
This is somehow an optimisation from the previous but, as the original version, it modifies the DNA.
When you call `adminWs.registerDna` you are changing the DNA (as far as I understand, maybe I'm wrong) by changing the `developer_address`

My initial try, which I couldn't achieve, was on line 40 `var agent_key = ...` to get the value from the DnaProperties
I believe we should not change the DNA but rather simulate who is connected